### PR TITLE
Build and parse json branch

### PR DESCRIPTION
**Pull request by moyaog**
The BuildJson class builds JSON objects that comply with JSON-RPC. The JSON object build will either be a request or a response. There will be no notifications.

The ParseJson class parses the JSON object received by either the client or the server. It expects an object built by a method in BuildJson that complies with JSON-RPC.

### CODE REVIEW

File: ParseJson.java [hunk 1]
```diff
@@ -12,10 +12,65 @@
       messageParams.put(METHOD, jsonObj.getString(METHOD));
     } 
     if(jsonObj.has(RESULT)) {
-      messageParams.put(RESULT, jsonObj.get(RESULT));
+      JSONObject res = jsonObj.getJSONObject(RESULT);
```
Comment (a-palchikov) [inline ParseJson.java:15]: I'm not sure how to deal with this code in this shape if I were to work with it - it looks pretty tangled and hardly readable.
Have you considered using a library for JSON (Jackson or Gson?) to help alleviate the complexities of unmarshalling and keeping the code straightforward?
If not, can you sketch a test for this functionality?

Comment (russjones) [reply ParseJson.java:15]: I second this, you really should not be doing any parsing of JSON yourself. If you need a schema, you can look into something [JSON Schema](https://json-schema.org) but that is also out of scope for this task.

Comment (moyaog) [reply ParseJson.java:15]: So I am not super familiar with JSON, but my understanding was that Jackson and Gson do not make JSON-RPC compliant JSON objects. Sasha and I discussed parsing in my first pull request here https://github.com/moyaog/PrototypeLinuxJobWorkerService/pull/1.

Comment (a-palchikov) [reply ParseJson.java:15]: I'm ok with this if this has been discussed and agreed upon.

Note though, JSON-RPC is **based** on JSON and there's nothing stopping you in producing or consuming objects with a JSON library in JSON-RPC format (which simply prescribes a specific field layout).

Comment (moyaog) [reply ParseJson.java:15]: Are there any other steps you would like me to take at at this point?

Comment (a-palchikov) [reply ParseJson.java:15]: You can merge this part as well.

**Review by a-palchikov — APPROVED** [review]

**Pull request merged**